lms/product/plugin/misc.py

```diff
@@ -111,16 +117,27 @@ def is_speed_grader_launch(self, _request) -> bool:  # pragma: nocover
 
     @staticmethod
     def _assignment_config_from_assignment(assignment: Assignment) -> AssignmentConfig:
-        return {
-            "document_url": assignment.document_url,
-            "group_set_id": assignment.extra.get("group_set_id"),
-        }
+        """Get the configuration of an assignment from our DB."""
+        config = AssignmentConfig(
+            document_url=assignment.document_url,
+            group_set_id=assignment.extra.get("group_set_id"),
+        )
+        if auto_grading_config := assignment.auto_grading_config:
+            config["auto_grading_config"] = auto_grading_config.asdict()
+
+        return config
 
     @staticmethod
     def _assignment_config_from_deep_linked_config(
         deep_linked_config: dict,
     ) -> AssignmentConfig:
-        return {
-            "document_url": deep_linked_config.get("url"),
-            "group_set_id": deep_linked_config.get("group_set"),
-        }
+        """Get the configuration of an assignment based on the Deep linked configuration."""
+        config = AssignmentConfig(
+            document_url=deep_linked_config.get("url"),
+            group_set_id=deep_linked_config.get("group_set"),
+        )
+
+        if auto_grading_config := deep_linked_config.get("auto_grading_config"):
+            config["auto_grading_config"] = json.loads(auto_grading_config)
```

Comment:
The raw deeplinkied value (a string) needs to be deserialized.

Reply:
Is it possible that the value may be invalid JSON? What happens in that case?

Reply:
I'm not going to say it's no possible but it would be very unexpected, this comes from the LMS and it's based on the parameters the backend provides on the deep-linking fields API call.

If this was not JSON I reckon the right approach is what we're doing here, just explode and get an exception on sentry.

If somehow we get this type of error from students ticknering with this APIs we could look into adding  validation but for now we haven't need it for any other fields (document, group set...)